mg.js

```diff
@@ -62,11 +62,30 @@ function createCardButton(frontImg) {
 
   return cardButton;
 }
-const shuffledFrontImgs = shuffleArray(doubledFrontImgs);
-const gameBoard = document.querySelector(".game-board");
-const cardContainer = document.createElement("div");
-cardContainer.className = "card-container";
-gameBoard.appendChild(cardContainer);
-shuffledFrontImgs.forEach((frontImg) => {
-  cardContainer.appendChild(createCardButton(frontImg));
+
+document.addEventListener("DOMContentLoaded", async function () {
+  let frontImg = [];
+  try {
+    const response = await fetch(
+      "https://raw.githubusercontent.com/SpringrollBrothers/SpringrollBrothers.github.io/main/memory.json"
+    );
+
+    if (!response.ok) {
+      throw new Error("Could not get your data");
+    }
+    const data = await response.json();
+    frontImg = data;
+
+    const doubledFrontImgs = [...frontImg, ...frontImg];
```

Comment:
This code works fine but it can become hard to maintain if you put all your application inside the `try / catch` statement. I recommend extracting the API call into a different function and have something like the following:

```js
async function getData() {
  try {
    // do the fetching
    return data;
  } catch (error) {
    // handle error
    return null;
  }
}

document.addEventListener("DOMContentLoaded", async function () {
  const frontImg = await getData();
  // ...
}
```